src/main/drivers/barometer/barometer_dps310.c

```diff
@@ -0,0 +1,321 @@
+
+    // 0x15 c10 [19:16] + 0x16 c10 [15:8] + 0x17 c10 [7:0]
+    baroState.calib.c10 = getTwosComplement((((uint32_t)coef[5] & 0x0F) << 16) | ((uint32_t)coef[6] << 8) | (uint32_t)coef[7], 20);
+
+    // 0x18 c01 [15:8] + 0x19 c01 [7:0]
+    baroState.calib.c01 = getTwosComplement(((uint32_t)coef[8] << 8) | (uint32_t)coef[9], 16);
+
+    // 0x1A c11 [15:8] + 0x1B c11 [7:0]
+    baroState.calib.c11 = getTwosComplement(((uint32_t)coef[8] << 8) | (uint32_t)coef[9], 16);
+
+    // 0x1C c20 [15:8] + 0x1D c20 [7:0]
+    baroState.calib.c20 = getTwosComplement(((uint32_t)coef[12] << 8) | (uint32_t)coef[13], 16);
+
+    // 0x1E c21 [15:8] + 0x1F c21 [7:0]
+    baroState.calib.c21 = getTwosComplement(((uint32_t)coef[14] << 8) | (uint32_t)coef[15], 16);
+
+    // 0x20 c30 [15:8] + 0x21 c30 [7:0]
+    baroState.calib.c30 = getTwosComplement(((uint32_t)coef[16] << 8) | (uint32_t)coef[17], 16);
+
+    // PRS_CFG: pressure measurement rate (32 Hz) and oversampling (16 time standard)
+    registerSetBits(busDev, DPS310_REG_PRS_CFG, DPS310_PRS_CFG_BIT_PM_RATE_32HZ | DPS310_PRS_CFG_BIT_PM_PRC_16);
+
+    // TMP_CFG: temperature measurement rate (32 Hz) and oversampling (16 times)
+    const uint8_t TMP_COEF_SRCE = registerRead(busDev, DPS310_REG_COEF_SRCE) & DPS310_COEF_SRCE_BIT_TMP_COEF_SRCE;
+    registerSetBits(busDev, DPS310_REG_TMP_CFG, DPS310_TMP_CFG_BIT_TMP_RATE_32HZ | DPS310_TMP_CFG_BIT_TMP_PRC_16 | TMP_COEF_SRCE);
+
+    // CFG_REG: set pressure and temperature result bit-shift (required when the oversampling rate is >8 times)
+    registerSetBits(busDev, DPS310_REG_CFG_REG, DPS310_CFG_REG_BIT_T_SHIFT | DPS310_CFG_REG_BIT_P_SHIFT);
+
+    // MEAS_CFG: Continuous pressure and temperature measurement
+    registerSetBits(busDev, DPS310_REG_MEAS_CFG, DPS310_MEAS_CFG_MEAS_CTRL_CONT);
+
+    return true;
+}
+
+static bool deviceReadMeasurement(baroDev_t *baro)
+{
+    // 1. Check if pressure is ready
+    bool pressure_ready = registerRead(baro->busDev, DPS310_REG_MEAS_CFG) & DPS310_MEAS_CFG_PRS_RDY;
+    if (!pressure_ready) {
+        return false;
+    }
+
+    // 2. Choose scaling factors kT (for temperature) and kP (for pressure) based on the chosen precision rate.
+    // The scaling factors are listed in Table 9.
+    static float kT = 253952; // 16 times (Standard)
+    static float kP = 253952; // 16 times (Standard)
+
+    // 3. Read the pressure and temperature result from the registers
+    // Read PSR_B2, PSR_B1, PSR_B0, TMP_B2, TMP_B1, TMP_B0
+    uint8_t buf[6];
+    if (!busReadBuf(baro->busDev, DPS310_REG_PSR_B2, buf, 6)) {
+        return false;
+    }
+
+    const int32_t Praw = getTwosComplement((buf[0] << 16) + (buf[1] << 8) + buf[2], 24);
+    const int32_t Traw = getTwosComplement((buf[3] << 16) + (buf[4] << 8) + buf[5], 24);
+
+    // 4. Calculate scaled measurement results.
+    const float Praw_sc = Praw / kP;
```

Comment:
I think this is causing the digit loss.

Reply:
Hmm... Still stepping ...
![Betaflight_Configurator](https://user-images.githubusercontent.com/14850998/85614596-0137b880-b696-11ea-8f18-2deb572936fa.jpg)